user/views.py

```diff
@@ -66,7 +66,7 @@ def current_user_view(request):
         if is_authenticated:
             return Response(UserSerializer(request.user).data)
         else:
-            return Response(success=False)
+            return Response(None)
```

Comment:
Wouldn't this return the response `{ "success": "true" }`?

Reply:
I tested it and it actually returns `{ "success" : "true", "data": null }` which works fine with the front-end auth code.

Reply:
Ok, I see what's happening. It's passing a null user with the login action in AuthProvider, which is treated the same as a logout (which is what happens with `success: false` anyways).

The one thing I'm wondering is if `success: false` should always correspond to an error condition. If so, it might be worth changing `UserViewSet.query` to throw an error if the user is not found.

Reply:
Added a change that makes it throw if you access it without being authenticated, which seems fine.

Reply:
Ahh. Hmm. I've been mistaken.

If an exception is raised, AuthProvider.check will throw it to the user, instead of logging out the user silently. We could also change the behaviour in the front end, but I think the `null` solution is more semantic.

Reply:
Alright, put it back to the way it was.